File: AddOns/Bookings/AccountKit/View/AKLabelCell.m
//
//  SGUserAccountStandardCell.m
//  WotGo
//
//  Created by Kuan Lun Huang on 12/12/2014.
//  Copyright (c) 2014 Adrian Schoenig. All rights reserved.
//

#import "AKLabelCell.h"

#ifdef TK_NO_FRAMEWORKS
#import "TripKit.h"
#import "TripKit/TripKit-Swift.h"
#else
@import TripKit;
@import TripKitUI;
#import "TripKitAddOns/TripKitAddOns-Swift.h"
#endif



@interface AKLabelCell ()

@end

@implementation AKLabelCell

@synthesize item = _item;

+ (AKLabelCell *)sharedInstance
{
  static AKLabelCell *_cell;
  
  static dispatch_once_t onceToken;
  
  dispatch_once(&onceToken, ^{
    _cell = [[self alloc] init];
  });
  
  return _cell;
}

#pragma mark - AKItemCell protocol

- (void)configureForItem:(AMKItem *)item
{
  ZAssert(item != nil, @"Item is missing");
  
  [self configureWithTitle:item.primaryText subtitle:item.secondaryText];
  
  if ([self shouldCenterTitle]) {
    [self adjustLayoutForCenteredTitle];
  }
  
  if (item.readOnly) {
    self.selectionStyle = UITableViewCellSelectionStyleNone;
  } else {
    self.selectionStyle = UITableViewCellSelectionStyleDefault;
  }
  
  _item = item;
}

#pragma mark - Public methods

- (void)configureWithTitle:(NSString *)title subtitle:(NSString *)subtitle
{
  _primaryLabel.text = title;
  _secondaryLabel.text = subtitle;
}

@end

File: AddOns/Bookings/AccountKit/View/AKLabelCell.swift
//
//  AMKLabelCell.swift
//  TripKit
//
//  Created by Kuan Lun Huang on 15/06/2015.
//  Copyright (c) 2015 SkedGo Pty Ltd. All rights reserved.
//

import UIKit

extension AKLabelCell {
  
  public func shouldCenterTitle() -> Bool {
    return self.primaryLabel?.text != nil && self.secondaryLabel?.text == nil;
  }
  
  public func adjustLayoutForCenteredTitle() {
    self.primaryLabel?.textAlignment = shouldCenterTitle() ? .center : .left
  }
  
  open override func updateConstraints() {
    self.primaryToSecondarySpacing?.constant = self.shouldCenterTitle() ? 0.0 : 8.0
    super.updateConstraints()
  }
  
  open override func prepareForReuse() {
    super.prepareForReuse()
    
    self.primaryLabel?.textAlignment = .left
    self.primaryToSecondarySpacing?.constant = 8.0
    self.primaryLabel?.textColor = SGStyleManager.darkTextColor()
    self.secondaryLabel?.textColor = SGStyleManager.lightTextColor()
  }
  
  open override func awakeFromNib() {
    super.awakeFromNib()
    
    self.primaryLabel?.textColor = SGStyleManager.darkTextColor()
    self.secondaryLabel?.textColor = SGStyleManager.lightTextColor()
  }
  
}
